x/lend/abci.go

```diff
@@ -0,0 +1,22 @@
+package lend
+
+import (
+	utils "github.com/comdex-official/comdex/types"
+	"github.com/comdex-official/comdex/x/lend/keeper"
+	"github.com/comdex-official/comdex/x/lend/types"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+func BeginBlocker(ctx sdk.Context, _ abci.RequestBeginBlock, k keeper.Keeper) {
+	defer telemetry.ModuleMeasureSince(types.ModuleName, ctx.BlockTime(), telemetry.MetricKeyBeginBlocker)
+
+	_ = utils.ApplyFuncIfNoError(ctx, func(ctx sdk.Context) error {
+		err := k.DeletePoolAndTransferFunds(ctx)
```

Comment:
method should be renamed to k.DeletePoolAndTransferInterest, as we are not transferring funds here, its the interest accrued been transferred.

Reply:
also I feel the should be executed at certain interval, instead of every block . thoughts ?